src/Validation/ValidatorAwareTrait.php

```diff
@@ -95,8 +95,16 @@ public function validator($name = null, Validator $validator = null)
         }
 
         if ($validator === null) {
-            $validator = new Validator();
-            $validator = $this->{'validation' . ucfirst($name)}($validator);
+            $className = isset($this->_validatorClass) ? $this->_validatorClass : 'Validator';
+            if ($className === 'Validator') {
+                $className = '\Cake\Validation\Validator';
+            }
+            $className = App::className($className, 'Validation/Validator', 'Validator');
```

Comment:
It took me a lot of effort to remove the dependency on `App` in the validation package. Can we just load by full class name?